gravitee-rest-api-management/gravitee-rest-api-management-rest/src/test/java/io/gravitee/rest/api/management/rest/resource/auth/OAuth2AuthenticationResourceTest.java

```diff
@@ -571,58 +572,49 @@ public void shouldConnectNewUserWithGroupsMappingFromUserInfo() throws Exception
-        verify(membershipService, times(1)).updateRoleToMemberOnReference(
-                new MembershipService.MembershipReference(MembershipReferenceType.GROUP, "group_id_2"),
-                new MembershipService.MembershipMember("janedoe@example.com", null, MembershipMemberType.USER),
-                new MembershipService.MembershipRole(RoleScope.APPLICATION, "ADMIN"),
-                USER_SOURCE_OAUTH2, false);
-
-        verify(membershipService, times(0)).updateRoleToMemberOnReference(
+        verify(membershipService, times(0)).updateRolesToMemberOnReference(
                 new MembershipService.MembershipReference(MembershipReferenceType.GROUP, "group_id_3"),
                 new MembershipService.MembershipMember("janedoe@example.com", null, MembershipMemberType.USER),
-                new MembershipService.MembershipRole(RoleScope.API, "USER"),
+                Arrays.asList(
+                        new MembershipService.MembershipRole(RoleScope.API, "USER"),
+                        new MembershipService.MembershipRole(RoleScope.APPLICATION, "ADMIN")
+                ),
                 USER_SOURCE_OAUTH2, false);
 
-        verify(membershipService, times(0)).updateRoleToMemberOnReference(
-                new MembershipService.MembershipReference(MembershipReferenceType.GROUP, "group_id_3"),
-                new MembershipService.MembershipMember("janedoe@example.com", null, MembershipMemberType.USER),
-                new MembershipService.MembershipRole(RoleScope.APPLICATION, "ADMIN"),
-                USER_SOURCE_OAUTH2, false);
-
-        verify(membershipService, times(1)).updateRoleToMemberOnReference(
-                new MembershipService.MembershipReference(MembershipReferenceType.GROUP, "group_id_4"),
-                new MembershipService.MembershipMember("janedoe@example.com", null, MembershipMemberType.USER),
-                new MembershipService.MembershipRole(RoleScope.API, "USER"),
-                USER_SOURCE_OAUTH2, false);
-
-        verify(membershipService, times(1)).updateRoleToMemberOnReference(
+        verify(membershipService, times(1)).updateRolesToMemberOnReference(
                 new MembershipService.MembershipReference(MembershipReferenceType.GROUP, "group_id_4"),
                 new MembershipService.MembershipMember("janedoe@example.com", null, MembershipMemberType.USER),
-                new MembershipService.MembershipRole(RoleScope.APPLICATION, "ADMIN"),
+                Arrays.asList(
+                        new MembershipService.MembershipRole(RoleScope.API, "USER"),
+                        new MembershipService.MembershipRole(RoleScope.APPLICATION, "ADMIN")
+                ),
                 USER_SOURCE_OAUTH2, false);
 
-        verify(membershipService, times(0)).updateRoleToMemberOnReference(
-                new MembershipService.MembershipReference(MembershipReferenceType.ENVIRONMENT, "DEFAULT"),
+        verify(membershipService, times(1)).updateRolesToMemberOnReference(
```

Comment:
We should have here 2 `verify`:
1. one for `MembershipReferenceType.ORGANIZATION` and role `MembershipRole(RoleScope.ORGANIZATION, "USER")`
2. one for `MembershipReferenceType.ENVIRONMENT` and role `MembershipRole(RoleScope.ENVIRONMENT, "ADMIN")`